notebooks/2023-01-17-initial-model-ph-mm-tl-kh/model_all.ipynb

```diff
@@ -0,0 +1,1156 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #1.    features.info()
</pre><p> </p><p>In the final training notebook, where the goal is not for experimentation anymore but rather the production of a model to be used for rollout, can we refactor the code to make it extremely easy to copy-paste the feature engineering steps? This minimizes the chances that the generation of features for rollout is different from how it was for training, given how we actually have a lot of ad-hoc steps scattered across different cells (dropping certain columns, retaining only columns with "_scaled", fillna with 0).</p><p> </p><p>E.g. Create a function that engineers the features for one set of geometries.&nbsp;So when you need to tweak/change it for training, you can just copy paste the whole function into the rollout notebook.</p><p> </p><p>Input: GDF of geometries</p><p>Output: GDF of geometries with features, and a list of feature column names</p><p>def generate_features(gdf, osm_year, ookla_year, ntl_year, scale_features=True)</p><p> </p><p><strong>Usage in the training notebook:</strong></p><pre class="ql-syntax" spellcheck="false">for each country:
&nbsp;&nbsp;dhs_gdf = generate_dhs_cluster_level_data()
&nbsp;&nbsp;dhs_gdf.drop_columns(drop_cols, inplace=True) # Drop the extra non-feature, non-label columns

   # This takes care of not only adding the features, but doing scaling and keeping only the appropriate columns, filling nans with 0. Essentially just gather all the steps needed to prepare the final features and put it in a function.
&nbsp;&nbsp;dhs_gdf = generate_features(dhs_gdf, osm_year, ookla_year, ntl_year, scale_values=True)

# Concatenate all the per country dhs_gdf
# Training procedure here on dhs_gdf
</pre><p> </p><p><strong>Usage in the rollout notebook:</strong></p><pre class="ql-syntax" spellcheck="false">bing_tiles = load_bing_tiles()

# The function definition is copy-pasted from the appropriate training notebook
bing_tiles = generate_features(bing_tiles, osm_year, ookla_year, ntl_year, scale_values=True) 

model = load_model()
model.predict(bing_tiles)
</pre><p> </p><p> </p><p>Suggesting this only because I think it'll be relatively quick and can be done in half a day. The idea is the same for both cross-country and single-country model training, and for rollout. </p><p> </p><p>Thoughts, agreement/disagreement? @jace, @jc, @butch</p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/thinkingmachines/unicef-ai4d-poverty-mapping/pull/139/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>